sdks/go/pkg/beam/core/runtime/harness/harness.go

```diff
@@ -506,6 +563,12 @@ func (c *control) handleInstruction(ctx context.Context, req *fnpb.InstructionRe
 func (c *control) getPlanOrResponse(ctx context.Context, kind string, instID, ref instructionID) (*exec.Plan, *metrics.Store, *fnpb.InstructionResponse) {
 	c.mu.Lock()
 	plan, ok := c.active[ref]
+	if !ok {
+		awaitingFinalization, ok := c.awaitingFinalization[ref]
```

Comment:
I'm not 100% sure, but I vaguely recall having an issue where using := in an inner scope would create a new variable of the same name as an outer scope variable. In this case, this inner ok would be a separate variable than the outer one, so like like lower down in line 579, the outer ok is still false and it would return a failure.

Regardless of whether this is intended behavior or not, I think you should rename this for clarity (something like ok2 is fine).

Reply:
Oh interesting - that makes sense, scoping is kinda weird in this situation.

Rather than using ok2, I think I prefer just not doing any initialization statements and updating to:

```
var af awaitingFinalization
af, ok = c.awaitingFinalization[ref]
if ok {
	plan = af.plan
}
```